nym-vpn-core/nym-vpn-lib/src/mobile/ios/default_path_observer.rs

```diff
@@ -0,0 +1,64 @@
+use tokio::sync::mpsc::{UnboundedReceiver, UnboundedSender};
+
+/// The type alias for the receiving end of path observer.
+pub type DefaultPathReceiver = UnboundedReceiver<OSDefaultPath>;
+
+/// Observer type that wraps network path changes into a channel.
+#[derive(Debug)]
+pub struct DefaultPathObserver {
```

Comment:
I like this, wrapping channels in a strong type. Imho we should do this more often

Reply:
Note that this struct implements `OSDefaultPathObserver` trait, making it a conduit for messages received via FFI boundary.